### ISSUE TITLE
GPU rendering on Linux inside VMware eats Windows host CPU, GPU

### ISSUE THREAD

**Issue by shoffmeister** [microsoft/vscode#129333]
We have written the needed data into your clipboard because it was too large to send. Please paste.
Issue Type: <b>Performance Issue</b>

I am running VS Code (Linux) inside VMware Workstation (Windows) - and every once in a while --type=gpu-process shows up on the command-line.

Whenever this is the case,
* Linux htop flags /usr/share/code with 35-50% of CPU usage (8 cores on the VM)
* Windows 10 Task Manager flags VMware Workstation consuming 25% CPU and 81%(!) GPU
* VS Code "Process Explorer" has the gpu-process at 4% CPU (and 0% for the broker child)

... while doing exactly nothing.

This has the effect that consumed power is high, with the net effect that
* this laptop's battery is eaten alive
* fans turn on, which gets 
* rather annoyingly loud, 
* making me less productive.

Some more context: The hardware is a three year-old HP zbook 15 with Intel integrated *and* nvidia dedicated built-in graphics. Attached is a Thunderbolt docking station with secondary (large) external screen.

In the issue scenario, it is the Intel GPU which is used by VMware Workstation and which gets pegged. Apparently load is sensitive to screen resolution - i.e. running VMware Workstation fullscreen (with Linux -> VS Code inside) on the large screen causes greater load than running it in a window, or on the FHD integrated laptop screen.

Software context: A managed Windows 10 Enterprise, 1909, with VMware Workstation (Player) 15.6, with 3D acceleration enabled; guest: Fedora 34, X11, KDE

VS Code version: Code 1.58.2 (c3f126316369cd610563c75b1b1725e0679adfb3, 2021-07-14T22:30:16.440Z)
OS version: Linux x64 5.13.4-200.fc34.x86_64
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2903)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|31.31GB (26.25GB free)|
|Process Argv|--unity-launch --crash-reporter-id 80e0ae97-0bc4-4a76-8337-f435f2ec1554|
|Screen Reader|no|
|VM|100%|
|DESKTOP_SESSION|/usr/share/xsessions/plasmax11|
|XDG_CURRENT_DESKTOP|KDE|
|XDG_SESSION_DESKTOP|KDE|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   160	  4365	code main
    0	    32	  4368	   zygote
    5	   128	  4397	     gpu-process
    0	     0	  4403	       broker
    0	    32	  4369	   zygote
    0	     0	  4371	     zygote
    0	    64	  7540	       window (Issue Reporter)
    0	    64	  4407	   utility
    5	   128	  4523	   shared-process
    0	    64	  4558	     ptyHost
    0	     0	  5230	       /usr/bin/bash
    0	     0	  7559	     /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    1	   257	  5170	   window (download-remote-config-files.sh (Untracked) - <masked> (Workspace) - Visual Studio Code)
    0	   128	  5196	     extensionHost
    0	    64	  5344	       /home/stefan/ManuallyInstalledSoftware/java/11/jdk-11.0.11+9/bin/java -noverify -Xmx64M -cp /home/stefan/.vscode/extensions/redhat.vscode-xml-0.17.0/server/org.eclipse.lemminx-0.17.1-uber.jar org.eclipse.lemminx.XMLServerLauncher
    0	    64	  6665	       /usr/share/code/code /usr/share/code/resources/app/extensions/json-language-features/server/dist/node/jsonServerMain --node-ipc --clientProcessId=5196
    0	   128	  5209	     watcherService
    0	    64	  5255	     searchService
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (download-remote-config-files.sh (Untracked) - <masked> (Workspace) - Visual Studio Code)
|    Folder (<masked>): 8759 files
|      File types: pyi(2271) py(2248) pyc(2234) so(193) json(118) txt(64)
|                  test(46) xml(43) h(40) jar(39)
|      Conf files: makefile(2) launch.json(1) settings.json(1)
|      Launch Configs: python;
```

</details>
<details><summary>Extensions (19)</summary>

Extension|Author (truncated)|Version
---|---|---
xml|Dot|2.5.1
gitlens|eam|11.6.0
rust-analyzer|mat|0.2.678
vscode-apache|mrm|1.2.0
vscode-docker|ms-|1.15.0
python|ms-|2021.7.1060902895
vscode-pylance|ms-|2021.7.5
jupyter|ms-|2021.8.1054968649
remote-containers|ms-|0.187.1
remote-ssh|ms-|0.65.7
remote-ssh-edit|ms-|0.65.7
remote-wsl|ms-|0.58.2
vscode-remote-extensionpack|ms-|0.21.0
powershell|ms-|2021.6.2
fabric8-analytics|red|0.3.3
vscode-commons|red|0.0.6
vscode-xml|red|0.17.0
vscode-yaml|red|0.21.1
vscode-lldb|vad|1.6.5

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
pythonvspyt700cf:30270857
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30311713
vspre833:30321513
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscus158:30321503
pythonvsuse255:30340121
vscod805:30301674
pythonvspyt200:30340761
vscextlang:30333561
binariesv615:30325510
pythonvssor306cf:30341832
bridge0708:30335490

```

</details>

**Comment by shoffmeister** [comment]
After haivng written the above, I noticed that the progress indicator for git operations in the status bar
![image](https://user-images.githubusercontent.com/3868036/126894893-cd53846e-10fe-4b92-ac43-f90215ee03ac.png)
was (still) turning.

I had successfully pushed to a remote ssh repo earlier - and then _noticed_ the fans spinning. Clicking on the progress indicator causes the command box to drop down, ESC, command box disappears, Windows task manager load goes down. So right now I g
